File: shopping/src/main/java/com/example/shopping/services/ManufacturerService.java
package com.example.shopping.services;

import com.example.shopping.model.entities.Client;
import com.example.shopping.model.entities.Manufacturer;
import org.springframework.stereotype.Service;

import java.util.List;

public interface ManufacturerService {
    Manufacturer saveManufacturer(Manufacturer manufacturer);
    List<Manufacturer> getAllManufacturers();
    Manufacturer getManufacturerById(Long id);
    Manufacturer disableManufacturer(Long id);
    Manufacturer activateManufacturer(Long id);
    Manufacturer updateManufacturerById (Manufacturer manufacturer);
}

File: shopping-angular/shop-angular/src/app/manufacturers/service/manufacturer.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Manufacturer } from '../model/manufacturer';
import { Observable, map } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class ManufacturerService {

  constructor(private httpClient: HttpClient) { }

  list(): Observable<Manufacturer[]>{
    return this.httpClient.get<Manufacturer[]>('http://localhost:8080/manufacturer');
  }

  saveManufacturer(manufacturer: Manufacturer): Observable<Manufacturer>{
    return this.httpClient.post<Manufacturer>('http://localhost:8080/manufacturer', manufacturer);
  }
  
  disableManufacturer(id: number): Observable<any> {
    const url =  'http://localhost:8080/manufacturer/disable/' + id
    return this.httpClient.put(url,null);
  }

  activateManufacturer(id: number): Observable<any> {
    const url =  'http://localhost:8080/manufacturer/activate/' + id
    return this.httpClient.put(url,null);
  }

  getManufacturerById(id: number): Observable<Manufacturer>{
    return this.httpClient.get('http://localhost:8080/manufacturer/' + id).pipe(
      map((data: any) => {
        // Transforme os dados para o tipo 'Client'
        return {
          id: data.id || null,
          name: data.name || '',
          createdAt: data.createdAt || null,
          updatedAt: data.updatedAt || null,
          active: data.active || false
        } as Manufacturer;
      })
    );
  }
}
